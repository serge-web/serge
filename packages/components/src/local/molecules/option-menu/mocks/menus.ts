import { faBath, faClone, faFileDownload, faTrash } from '@fortawesome/free-solid-svg-icons'

export default [
  {
    onClick: (): void => window.alert('Click on menu #1'),
    icon: faBath,
    title: 'Save clean copy of wargame (no messages)',
    label: 'Clean copy'
  },
  {
    onClick: (): void => window.alert('Click on menu #2'),
    icon: faClone,
    title: 'Duplicate wargame',
    label: 'Duplicate'
  },
  {
    onClick: (): void => window.alert('Click on menu #3'),
    icon: faFileDownload,
    title: 'Export wargame',
    label: 'Export'
  },
  {
    onClick: (): void => window.alert('Click on menu #4'),
    icon: faTrash,
    title: 'Delete wargame',
    label: 'Delete'
  }
]
