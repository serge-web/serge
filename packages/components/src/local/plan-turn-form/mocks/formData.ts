const formData = {
  populate: {
    status: [{
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }],
    speed: [10, 20, 30]
  },
  values: {
    statusVal: {
      name: 'Transiting',
      mobile: true
    },
    speedVal: 10,
    turnsVal: 5,
    condition: 'Working'
  }
}

export default formData
