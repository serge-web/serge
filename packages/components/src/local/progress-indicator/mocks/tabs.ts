export const tabs = [
  {
    name: 'overview',
    complete: false
  },
  {
    name: 'forces',
    complete: false
  },
  {
    name: 'channels',
    complete: false
  }
]

export const tabsVariant = [
  {
    name: 'overview',
    complete: true
  },
  {
    name: 'forces',
    complete: false
  },
  {
    name: 'channels',
    complete: false
  }
]

export default tabs
