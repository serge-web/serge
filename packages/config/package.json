{
  "name": "@serge/config",
  "version": "0.3.0",
  "license": "Apache-2.0",
  "author": "foxleigh81 <alex@alexfoxleigh.com>",
  "main": "build/index.js",
  "dependencies": {
    "@types/dotenv": "^8.2.0",
    "@types/expired-storage": "^1.0.0",
    "dotenv": "^8.2.0",
    "expired-storage": "^1.0.2",
    "typescript": "^4.0.5"
  },
  "scripts": {
    "build": "tsc --pretty",
    "watch": "tsc --watch"
  }
}
