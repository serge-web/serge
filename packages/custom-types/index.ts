/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as Asset } from './asset.d'
export * from './attributes'
export * from './channel-data.d'
export { default as ChannelUI } from './channel-ui.d'
export { default as Channel } from './channel.d'
export { default as Editor } from './editor.d'
export * from './export-items-actions.d'
export * from './export-items.d'
export { default as ForceData } from './force-data.d'
export { default as ForceOption } from './force-option.d'
export { default as ForceRole } from './force-role.d'
export { default as Force } from './force.d'
export * from './forms.d'
export * from './global.d'
export { default as GroupItem } from './group-item.d'
export { default as IconOption } from './icon-option.d'
export * from './map-annotation'
export { default as MapPostBack } from './map-postback-data.d'
export { default as MappingConstraints } from './mapping-constraints.d'
export { default as MappingContext } from './mapping-context.d'
export { default as MarkerPerception } from './marker-perception.d'
export * from './message-types-actions.d'
export { default as MessageTypes, TemplateBodysByKey } from './message-types.d'
export {
  ChatMessage, CollaborationDetails, CoreMessage, default as Message, FeedbackItem, InteractionDetails
  // All template based messages
  , InteractionMessageStructure, MessageChannel, MessageCloneMarker, MessageCounter, MessageCreateTaskGroup,
  // All template based messages
  MessageCustom, MessageDeleteMarker, MessageDeletePlatform, MessageDetails,
  MessageDetailsFrom,
  // System Messages
  MessageFeedback,
  // Map Types
  MessageForceLaydown, MessageHostPlatform, MessageInfoType,
  MessageInfoTypeClipped, MessageInteraction, MessageLeaveTaskGroup, MessageMap, MessagePerceptionOfContact, MessagePlanning,
  MessageStateOfWorld, MessageStructure, MessageSubmitPlans, MessageUpdateMarker, MessageVisibilityChanges, PlanningMessageStructure
} from './message.d'
export { MessagesActionTypes, MessagesDispatch, SetCurrentViewFromURI } from './messages-action'
export * from './participant.d'
export { default as PerceivedTypes } from './perceived-types.d'
export { default as Perception } from './perception.d'
export { default as PlanMobileAsset, TurningDetails } from './plan-mobile-asset'
export { default as PlannedRoute } from './planned-route.d'
export * from './planning-activity'
export { default as PlatformTypeData } from './platform-type-data.d'
export { default as PlatformType } from './platform-type.d'
export * from './player-log-actions'
export * from './player-log.d'
export * from './player-ui-actions.d'
export { default as PlayerUi, PlayerUiChannels, PlayerUiChatChannel } from './player-ui.d'
export { default as RequestForInformation } from './request-for-information.d'
export { default as Role } from './role.d'
export * from './root-state'
export { default as RouteMarker, RouteMarkerDuo, RouteMarkerStatus as RouteMarkerStatus } from './route-marker'
export { default as RouteStatus } from './route-status.d'
export { default as RouteStore } from './route-store.d'
export { default as RouteTurn } from './route-turn'
export { default as Route, RouteChild } from './route.d'
export { default as SelectedAsset } from './selected-asset.d'
export * from './serge-hex-3.d'
export { default as SetWargameMessage } from './set-wargame-message.d'
export { AssetState, ForceState, StateOfWorld } from './state-of-world.d'
export { default as State } from './state.d'
export { default as Template, TemplateBody } from './template.d'
export * from './turn-length.d'
export { default as Visibility } from './visibility.d'
export { WargameActionTypes, WargameDispatch, WargameRevision } from './wargame-action'
export { default as WargameData } from './wargame-data.d'
export { default as WargameList } from './wargame-list.d'
export { default as WargameOverview } from './wargame-overview.d'
export { default as Wargame } from './wargame.d'

