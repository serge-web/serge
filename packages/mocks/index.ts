/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as GameMessagesMock } from './game-messages.mock'
export { default as GameMessagesMockRFI } from './game-messages-rfi.mock'
export { default as AdminMessagesMock } from './admin-messages.mock'
export { default as ChatMessagesMock } from './chat-messages.mock'
export { default as infoMarker } from './map-annotation.mock'
export { default as InfoMessagesMock } from './info-messages.mock'
export { default as annotationMarkerData } from './annotation-marker-data.mosk'
export { default as GameChannels2 } from './game-channels-2.mock'
export { default as WargameMock } from './wargame.mock'
export { default as MapAnnotationMock } from './map-annotation.mock'
export { default as WargameExportedMock } from './wargame-exported.mock'
export { default as P9Mock } from './p9.mock'
export { default as MessageTemplatesMock } from './message-templates.mock'
export { default as MessageTemplatesMockByKey } from './message-templates-by-key.mock'
export { default as localMappingConstraints } from './local-mapping.mock'
export { default as forces } from './forces.mock'
export { default as atlanticForces } from './atlantic-forces.mock'
export { default as smallForces } from './small-forces.mock'
export { default as smallScaleForces } from './small-scale-forces.mock'
export { default as watuWargame } from './watu-wargame-mock'
export { default as watuPlaytest } from './watu-playtest-mock'
export { default as p9wargame } from './p9.mock'
export { default as cmdWkWargame } from './cmd-week.mock'
export { default as smallScaleMappingConstraints } from './small-scale-mapping-constraints.mock'
export { default as preInitialisedForces } from './pre-initialised-forces.mock'
export { default as platformTypes } from './platform-types.mock'
export { default as platformType } from './platform-type.mock'
export { default as selectedAsset } from './selected-asset.mock'
export { default as adminTabs } from './admin-tabs.mock'
export { default as feedbackMessage, feedbackMessage1, feedbackMessage2 } from './feedback-message.mock'
export { default as actionFeedbackMessages } from './action-feedback-messages.mock'
export { default as actionSetLatestFeedbackMessage } from './action-set-latest-feedback-message.mock'
export { default as actionSetLatestWargameMessage } from './action-set-latest-wargame-message.mock'
export { default as actionSetWargameMessages } from './action-set-wargame-messages.mock'
export { default as actionOpenMessage } from './action-open-message.mock'
export { default as actionCloseMessage } from './action-close-message.mock'
export * from './coa-channel-data-2.mock'
export * from './planning-messages.mock'
export { default as planningMessageTemplatesMock } from './planning-message-templates.mock'
export * from './coa-messages.mock'
