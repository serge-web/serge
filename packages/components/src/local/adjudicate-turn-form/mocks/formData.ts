import { AdjudicateTurnFormPopulate, AttributeTypes } from '@serge/custom-types'

const attributes: AttributeTypes = []

const formData: AdjudicateTurnFormPopulate = {
  contactId: 'C234',
  status: [
    {
      name: 'Fishing',
      mobile: false
    },
    {
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }
  ],
  speed: [10, 20, 30],
  visibleTo: [
    {
      name: 'Blue Force',
      colour: '#69c',
      id: 'Blue'
    },
    {
      name: 'Red Force',
      colour: '#f00',
      id: 'Red'
    },
    {
      name: 'White Force',
      colour: '#fff',
      id: 'umpire'
    }
  ],
  condition: ['Working', 'Disabled', 'Immobile', 'Destroyed'],
  attributes: attributes
}

export default formData
