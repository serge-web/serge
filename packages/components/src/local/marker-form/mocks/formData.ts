import { PerceptionFormData } from '@serge/custom-types'

const formData: PerceptionFormData = {
  populate: {
    perceivedForces: [
      {
        name: 'Blue Force',
        colour: '#69c',
        id: 'Blue'
      },
      {
        name: 'Red Force',
        colour: '#f00',
        id: 'Red'
      },
      {
        name: 'White Force',
        colour: '#fff',
        id: 'Umpire'
      }
    ],
    perceivedTypes: [
      { uniqid: 'aa', name: 'Fishing vessel' },
      { uniqid: 'a1', name: 'torpedo' },
      { uniqid: 'a2', name: 'Boghammer' },
      { uniqid: 'a3', name: 'Frigate' }
    ]
  },
  values: {
    perceivedNameVal: 'Test',
    perceivedForceId: 'Blue Force',
    perceivedForceColor: '#69c',
    perceivedTypeId: 'a3',
    assetId: 'c00012',
    iconURL: 'frigate.svg'
  }
}

export default formData
