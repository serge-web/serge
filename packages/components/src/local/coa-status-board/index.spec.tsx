/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'
import { ForceRole, MessageCustom, Role, ChannelData } from '@serge/custom-types'
import { CUSTOM_MESSAGE, CollaborativeMessageStates, SpecialChannelTypes } from '@serge/config'
import { forces, MessageTemplatesMockByKey, WargameMock } from '@serge/mocks'
import CoaStatusBoard from './index'
import moment from 'moment'

/* Import mock data */
// import { GameChannels } from '@serge/mocks'

const whiteGC: Role = forces[0].roles[0]
const blueCO: Role = forces[1].roles[0]
const redCO: Role = forces[2].roles[0]

/* TODO: test data */
const testChannel: ChannelData = {
  collabOptions: {
    extraColumns: [],
    mode: 'edit',
    returnVerbs: ['Endorse', 'Request Changes'],
    startWithReview: true
  },
  format: SpecialChannelTypes.CHANNEL_COLLAB_EDIT,
  name: 'New CE',
  participants: [
    {
      canCollaborate: true,
      canReleaseMessages: false,
      force: 'White',
      forceUniqid: 'umpire',
      roles: [],
      subscriptionId: 'oqoj',
      templates: [
        { title: 'RFI', _id: 'k16eedkj' }
      ]
    }
  ],
  uniqid: 'ks8soryj'
}

/* TODO: test data */
const testMock: MessageCustom[] = [
  {
    messageType: CUSTOM_MESSAGE,
    details: {
      channel: 'channel-BlueRFI',
      from: {
        force: 'Blue',
        forceColor: '#00F',
        iconURL: 'default_img/umpireDefault.png',
        roleName: blueCO.name,
        roleId: blueCO.roleId
      },
      messageType: 'RFI',
      timestamp: '2020-10-13T08:54:21.119Z',
      collaboration: {
        lastUpdated: moment(new Date(), moment.ISO_8601).format(),
        status: CollaborativeMessageStates.Pending
      },
      turnNumber: 1
    },
    message: {
      Reference: 'Blue-3',
      Request: 'RFI 3 request goes in here',
      Title: 'RFI 3 from Blue'
    },
    _id: 'id_3a',
    _rev: '2',
    hasBeenRead: false,
    isOpen: false
  },
  {
    messageType: CUSTOM_MESSAGE,
    details: {
      channel: 'Red RFI',
      from: {
        force: 'Red',
        roleName: redCO.name,
        forceColor: '#F00',
        iconURL: 'default_img/umpireDefault.png',
        roleId: redCO.roleId
      },
      messageType: 'RFI',
      timestamp: '2020-10-13T08:55:21.119Z',
      collaboration: {
        lastUpdated: moment(new Date(), moment.ISO_8601).format(),
        status: CollaborativeMessageStates.BeingEdited
      },
      turnNumber: 1
    },
    message: {
      Reference: 'RED-1',
      Request: 'RFI 1 request from red goes in here',
      Title: 'RFI 1 from RED'
    },
    _rev: '1',
    _id: 'id_4',
    hasBeenRead: false,
    isOpen: false
  },
  {
    messageType: CUSTOM_MESSAGE,
    details: {
      channel: 'channel-k63pjit0',
      from: {
        force: 'White',
        forceColor: '#FCFBEE',
        iconURL: 'default_img/umpireDefault.png',
        roleName: whiteGC.name,
        roleId: whiteGC.roleId
      },
      messageType: 'Chat',
      privateMessage: 'The private content goes in here',
      timestamp: '2020-10-13T08:52:40.930Z',
      collaboration: {
        lastUpdated: moment(new Date(), moment.ISO_8601).format(),
        status: CollaborativeMessageStates.Closed,
        response: 'Game control response to RFI 4'
      },
      turnNumber: 1
    },
    message: {
      content: 'Message from White, with Private content'
    },
    _id: '2020-03-25T15:08:47.540Z',
    _rev: '1',
    hasBeenRead: false,
    isOpen: false
  },
  {
    messageType: CUSTOM_MESSAGE,
    details: {
      channel: 'channel-k63pjit0',
      from: {
        force: 'Red',
        forceColor: '#F00',
        roleName: redCO.name,
        iconURL: 'default_img/umpireDefault.png',
        roleId: redCO.roleId
      },
      messageType: 'Chat',
      timestamp: '2020-10-13T08:52:04.394Z',
      collaboration: {
        lastUpdated: moment(new Date(), moment.ISO_8601).format(),
        status: CollaborativeMessageStates.PendingReview,
        response: 'Game control response to RFI 4'
      },
      turnNumber: 1
    },
    message: {
      content: 'message from Red'
    },
    _id: '2020-03-25T15:08:47.525Z',
    _rev: '1',
    hasBeenRead: false,
    isOpen: false
  },
  {
    messageType: CUSTOM_MESSAGE,
    details: {
      channel: 'channel-k63pjit0',
      from: {
        force: 'Blue',
        forceColor: '#00F',
        roleName: blueCO.name,
        iconURL: 'default_img/umpireDefault.png',
        roleId: blueCO.roleId
      },
      messageType: 'Chat',
      timestamp: '2020-10-13T08:52:21.119Z',
      collaboration: {
        lastUpdated: moment(new Date(), moment.ISO_8601).format(),
        status: CollaborativeMessageStates.Released,
        response: 'Game control response to RFI 4'
      },
      turnNumber: 1
    },
    message: {
      content: 'Message from Blue'
    },
    _id: '2020-03-25T15:08:47.530Z',
    _rev: '1',
    hasBeenRead: false,
    isOpen: false
  }
]

const role: ForceRole = { forceId: 'umpire', forceName: 'White', roleId: 'game control', roleName: 'Game Control' }

const onChange = (): void => console.log()

it('CoaStatusBoard renders correctly', () => {
  const handleReadMessage = (message: MessageCustom): void => {
    console.log('message read: ' + message)
  }

  const tree = renderer
    .create(
      <CoaStatusBoard
        currentWargame='wargame-test'
        templates={MessageTemplatesMockByKey}
        forces={forces}
        role={role}
        isUmpire={true}
        channel={testChannel}
        messages={testMock}
        onChange={onChange}
        gameDate={WargameMock.data.overview.gameDate}
        onMessageRead={handleReadMessage}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
