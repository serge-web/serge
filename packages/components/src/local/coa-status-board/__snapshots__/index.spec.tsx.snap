// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoaStatusBoard renders correctly 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root makeStyles-tableHead-2"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <span>
            ID
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              From
              <span
                className="MuiButton-endIcon MuiButton-iconSizeSmall"
              >
                <svg
                  fill="none"
                  height="30"
                  viewBox="0 0 28 30"
                  width="28"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M26.2096 0H1.29056C0.145551 0 -0.432218 1.51559 0.379086 2.40064L10.3125 13.2387V25.3125C10.3125 25.7713 10.5177 26.2014 10.8623 26.4646L15.1592 29.7446C16.0069 30.392 17.1875 29.7359 17.1875 28.5926V13.2387L27.1212 2.40064C27.9308 1.51734 27.357 0 26.2096 0Z"
                    fill="white"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <span>
            Title
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Status
              <span
                className="MuiButton-endIcon MuiButton-iconSizeSmall"
              >
                <svg
                  fill="none"
                  height="30"
                  viewBox="0 0 28 30"
                  width="28"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M26.2096 0H1.29056C0.145551 0 -0.432218 1.51559 0.379086 2.40064L10.3125 13.2387V25.3125C10.3125 25.7713 10.5177 26.2014 10.8623 26.4646L15.1592 29.7446C16.0069 30.392 17.1875 29.7359 17.1875 28.5926V13.2387L27.1212 2.40064C27.9308 1.51734 27.357 0 26.2096 0Z"
                    fill="white"
                  />
                </svg>
              </span>
            </span>
          </button>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          <span>
            Owner
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root makeStyles-tableBody-3"
      role={null}
    >
      <tr
        className="MuiTableRow-root makeStyles-tableRow-4 makeStyles-tableRowCollapsibleTrigger-6"
        onClick={[Function]}
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <svg
            className="svg-inline--fa fa-plus"
          />
           
          Blue-3
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-12"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-13"
            >
              CO
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          RFI 3 from Blue
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-14"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-15"
            >
              Response Unallocated
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-16"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-17"
            >
              Pending
            </span>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRowCollapsible-5"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          colSpan={5}
        >
          <div
            className="MuiCollapse-container MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRow-4 makeStyles-tableRowCollapsibleTrigger-6"
        onClick={[Function]}
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <svg
            className="svg-inline--fa fa-plus"
          />
           
          RED-1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-18"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-19"
            >
              CO
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          RFI 1 from RED
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-20"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-21"
            >
              Edit Response
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-22"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-23"
            >
              Pending
            </span>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRowCollapsible-5"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          colSpan={5}
        >
          <div
            className="MuiCollapse-container MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRow-4 makeStyles-tableRowCollapsibleTrigger-6"
        onClick={[Function]}
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <svg
            className="svg-inline--fa fa-plus"
          />
           
          2020-03-25T15:08:47.540Z
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-24"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-25"
            >
              Game Control
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        />
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-26"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-27"
            >
              Closed
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-28"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-29"
            >
              Pending
            </span>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRowCollapsible-5"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          colSpan={5}
        >
          <div
            className="MuiCollapse-container MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRow-4 makeStyles-tableRowCollapsibleTrigger-6"
        onClick={[Function]}
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <svg
            className="svg-inline--fa fa-plus"
          />
           
          2020-03-25T15:08:47.525Z
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-30"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-31"
            >
              CO
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        />
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-32"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-33"
            >
              Pending review
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-34"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-35"
            >
              Pending
            </span>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRowCollapsible-5"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          colSpan={5}
        >
          <div
            className="MuiCollapse-container MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRow-4 makeStyles-tableRowCollapsibleTrigger-6"
        onClick={[Function]}
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <svg
            className="svg-inline--fa fa-plus"
          />
           
          2020-03-25T15:08:47.530Z
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-36"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-37"
            >
              CO
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        />
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-38"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-39"
            >
              Released
            </span>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="MuiChip-root makeStyles-root-9 makeStyles-root-40"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label makeStyles-label-11 makeStyles-label-41"
            >
              Pending
            </span>
          </div>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root makeStyles-tableRowCollapsible-5"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
          colSpan={5}
        >
          <div
            className="MuiCollapse-container MuiCollapse-hidden"
            style={
              Object {
                "minHeight": "0px",
              }
            }
          >
            <div
              className="MuiCollapse-wrapper"
            >
              <div
                className="MuiCollapse-wrapperInner"
              />
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
